table/block_based/block_based_table_factory.cc

```diff
@@ -782,6 +783,15 @@ std::string BlockBasedTableFactory::GetPrintableOptions() const {
     }
     ret.append("  block_cache_options:\n");
     ret.append(table_options_.block_cache->GetPrintableOptions());
+    if (block_cache_name != nullptr &&
+        std::strcmp(block_cache_name, "LRUCache") == 0) {
+      LRUCache* lru_cache =
+          dynamic_cast<LRUCache*>(table_options_.block_cache.get());
+      if (lru_cache->GetSecondaryCache()) {
+        ret.append("  secondary cache:\n");
+        ret.append(lru_cache->GetSecondaryCache()->GetPrintableOptions());
+      }
+    }
```

Comment:
Instead of  this, why not have the LRUCache::GetPrintableOptions invoked secondary cache::GetPrintableOptions?

Reply:
Thanks for this recommendation! I prepared a new commit for this approach.